storage/s3/src/common/Helpers.ts

```diff
@@ -0,0 +1,94 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { S3Client } from "@aws-sdk/client-s3";
+import { STSClient } from "@aws-sdk/client-sts";
+
+import {
+  assertTransferConfig,
+  TransferConfig,
+} from "@itwin/object-storage-core/lib/common";
+
+import {
+  assertPrimitiveType,
+  FalsyValueError,
+} from "@itwin/cloud-agnostic-core";
+
+import { S3TransferConfig } from "./Interfaces";
+
+export function assertS3TransferConfig(
+  transferConfig: TransferConfig | S3TransferConfig
+): asserts transferConfig is S3TransferConfig {
+  assertTransferConfig(transferConfig);
+  if (!("authentication" in transferConfig))
+    throw new FalsyValueError("transferConfig.authentication");
+
+  assertPrimitiveType(
```

Comment:
This is an example of where it's fine to use visual grouping with whitespace. It's within a single code block and helps with readability. These `assertPrimitiveType` could be split up by empty lines or be left as is, but for other parts of this method there's no question that you can't add empty lines between statements.